src/dft/backends/mklgpu/commit.cpp

```diff
@@ -23,118 +23,98 @@
         // MKLGPU does not throw an informative exception for the following:
         if constexpr (prec == dft::detail::precision::DOUBLE) {
             if (!queue.get_device().has(sycl::aspect::fp64)) {
-                throw mkl::exception("dft/backends/mklgpu", "commit",
-                                     "Device does not support double precision.");
+                throw math::exception("dft/backends/mklgpu", "commit",
+                                      "Device does not support double precision.");
             }
         }
     }
 
     virtual void commit(const dft::detail::dft_values<prec, dom>& config_values) override {
         this->external_workspace_helper_ =
-            oneapi::mkl::dft::detail::external_workspace_helper<prec, dom>(
+            oneapi::math::dft::detail::external_workspace_helper<prec, dom>(
                 config_values.workspace_placement ==
-                oneapi::mkl::dft::detail::config_value::WORKSPACE_EXTERNAL);
+                oneapi::math::dft::detail::config_value::WORKSPACE_EXTERNAL);
 
         auto stride_choice = dft::detail::get_stride_api(config_values);
         throw_on_invalid_stride_api("MKLGPU commit", stride_choice);
         // A separate descriptor for each direction may not be required.
         bool one_descriptor = (stride_choice == dft::detail::stride_api::FB_STRIDES) ||
                               (config_values.input_strides == config_values.output_strides);
-        bool forward_good = true;
         // Make sure that second is always pointing to something new if this is a recommit.
         handle.second = handle.first;
 
         // Generate forward DFT descriptor. If using FWD/BWD_STRIDES API, only
         // one descriptor is needed.
         set_value(*handle.first, config_values, true, stride_choice);
-        try {
-            handle.first->commit(this->get_queue());
-        }
-        catch (const std::exception& mkl_exception) {
-            // Catching the real Intel oneMKL exception causes headaches with naming
-            forward_good = false;
-            if (one_descriptor) {
-                throw mkl::exception("dft/backends/mklgpu"
-                                     "commit",
-                                     mkl_exception.what());
-            }
-        }
+        RETHROW_ONEMKL_EXCEPTIONS(handle.first->commit(this->get_queue()));
 
         // Generate backward DFT descriptor only if required.
         if (!one_descriptor) {
             handle.second = std::make_shared<mklgpu_descriptor_t>(config_values.dimensions);
             set_value(*handle.second, config_values, false, stride_choice);
-            try {
-                handle.second->commit(this->get_queue());
-            }
-            catch (const std::exception& mkl_exception) {
-                // Catching the real Intel oneMKL exception causes headaches with naming.
-                if (!forward_good) {
```

Comment:
If we are only using the `fwd/bwd_strides` then its fine to clean up this logic, I don't think this change is a risk as we want to use the strides config anyway and move away from the `input/output_distance`. Perhaps it would be wise to run the DFT GPU tests for this PR as well (as I only see only the CPU unit tests have run in post-commit.

Reply:
Ok, good. The logs are attached in the PR's description. One of them is running on PVC. There are some failures that can also be reproduced on the develop branch.